### PR TITLE
Add DeviceAddResult

### PR DESCRIPTION
**Pull request by giarc3**
- Added `DeviceAddResult`, the new result of `generate_new_device()`
- Made changes to optional name fields' constructors (like `GroupName`, `DeviceName`, etc.) for consistency. _Please check if these are reasonable_.

TODO:
- [x] Depend on released ironoxide-java

### CODE REVIEW

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.ironcorelabs.scala.sdk
+
+import com.ironcorelabs.{sdk => jsdk}
+import java.{util => ju}
+
+/**
+ * Account's device context. Needed to initialize the SDK with a set of device keys.
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: This comment seems wrong. It's the AddResult.

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+case class DeviceAddResult(
+  userId: UserId,
+  segmentId: Long,
+  devicePrivateKey: PrivateKey,
+  signingPrivateKey: DeviceSigningPrivateKey,
+  deviceId: DeviceId,
+  name: Option[DeviceName],
+  created: ju.Date,
+  lastUpdated: ju.Date
+)
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: Should this case class have a `toDeviceContext` to represent the easy conversion?

Comment (giarc3) [reply src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: I was thinking a constructor for `DeviceContext` that takes a `DeviceAddResult` would be cleaner. And I thought I had done that but obviously I'm crazy.
Which way would you prefer? (or both?)

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala (hunk 1 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: Good catch

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala (hunk 2 above)
Comment (coltfred) [reply src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: Both sounds great.

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala (hunk 1 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: Fixed.

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala (hunk 2 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: Added both

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  userId: UserId,
+  segmentId: Long,
+  devicePrivateKey: PrivateKey,
+  signingPrivateKey: DeviceSigningPrivateKey,
+  deviceId: DeviceId,
+  name: Option[DeviceName],
+  created: ju.Date,
+  lastUpdated: ju.Date
+) {
+  def toDeviceContext(): DeviceContext =
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: Don't need the `()`

File: src/test/scala/ironoxide/FullIntegrationTest.scala [hunk 4]
```diff
@@ -52,8 +53,8 @@ class FullIntegrationTest extends AsyncWordSpec with Matchers with EitherValues
 
   "DeviceCreateOpts" should {
     "create with DeviceName of null" in {
```
Comment (coltfred) [inline src/test/scala/ironoxide/FullIntegrationTest.scala]: The name of this test should change now to indicate that it's none, not null.

File: src/main/scala/com/ironcorelabs/DeviceAddResult.scala (hunk 3 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/DeviceAddResult.scala]: I blame Rust. Fixed.

File: src/test/scala/ironoxide/FullIntegrationTest.scala (hunk 4 above)
Comment (giarc3) [reply src/test/scala/ironoxide/FullIntegrationTest.scala]: Changed.

**Pull request merged**